app/Http/Controllers/DiscountController.php

```
@@ -0,0 +1,161 @@
+        // Validate percentage value
+        if ($validated['type'] === 'percent' && $validated['value'] > 10000) {
+            return redirect()->back()
+                ->with('error', 'Percentage discount cannot exceed 100% (10000 basis points).')
+                ->withInput();
+        }
+
+        $validated['user_id'] = auth()->id();
+        $validated['active'] = $request->has('active');
+        $validated['redemptions_count'] = 0;
+
+        $discount = Discount::create($validated);
```

PHPMD (2.15.0)

Avoid unused local variables such as '$discount'. (undefined)

(UnusedLocalVariable)